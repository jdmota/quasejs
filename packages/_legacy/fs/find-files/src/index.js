const fs = require( "fs-extra" );
const path = require( "path" );
const ignoredDirectories = require( "ignore-by-default" ).directories().map( dir => `!**/${dir}/**` );
const mm = require( "micromatch" );
const globParent = require( "glob-parent" );
const toAbsoluteGlob = require( "to-absolute-glob" );
const isNegatedGlob = require( "is-negated-glob" );
const slash = require( "slash" );
const Observable = require( "zen-observable" );

class Pattern {

  constructor( _pattern, { cwd, micromatch } ) {
    const micromatchOpts = this.negated ? Object.assign( {}, micromatch, { dot: true } ) : micromatch;
    const { negated, pattern } = isNegatedGlob( _pattern );
    this.negated = negated;
    this.regexp = mm.makeRe( pattern, micromatchOpts );
    this.endsWithGlobstar = pattern.slice( -3 ) === "/**";
    this.parent = globParent( pattern ).replace( /^\.$/g, "" );
    this.fullParent = toAbsoluteGlob( this.parent, { cwd } );
  }

}

class Reader {

  constructor( patterns, observer, options ) {
    this.patterns = patterns;
    this.cwd = options.cwd;
    this.observer = observer;
    this.cancelled = false;
    this.pending = 0;
    this.includeDirs = options.includeDirs;
    this.visitedDirs = options.visitedDirs;
    this.relative = options.relative;
    this.fsStat = options.fs.stat;
    this.fsReadDir = options.fs.readdir;
    this.error = this.error.bind( this );
    this.cancel = this.cancel.bind( this );
  }

  start() {
    for ( const { negated, fullParent } of this.patterns ) {
      if ( !negated ) {
        this.check( fullParent );
      }
    }
    this.checkPending();
    return this.cancel;
  }

  checkPending() {
    if ( this.pending === 0 ) {
      this.complete();
    }
  }

  next( file ) {
    this.observer.next( file );
  }

  complete() {
    this.observer.complete();
  }

  error( error ) {
    this.observer.error( error );
  }

  cancel() {
    this.cancelled = true;
  }

  // We start from the end
  // negated matches
  //    0      0     keep going
  //    0      1     return true
  //    1      0     keep going
  //    1      1     return false
  matches( string ) {
    const patterns = this.patterns;
    let i = patterns.length;
    while ( i-- ) {
      const negated = patterns[ i ].negated;
      const matches = patterns[ i ].regexp.test( string );
      if ( matches ) {
        return !negated;
      }
    }
    return false;
  }

  matchesDir( dir ) {
    const patterns = this.patterns;
    let i = patterns.length;
    while ( i-- ) {
      const p = patterns[ i ];
      if ( p.negated ) {
        if ( p.endsWithGlobstar ) {
          const matches = p.regexp.test( dir + "/" );
          if ( matches ) {
            return false;
          }
        }
      } else {
        const parent = p.parent;
        let matches = true;
        const len = Math.min( parent.length, dir.length );
        for ( let i = 0; i < len; i++ ) {
          if ( parent.charCodeAt( i ) !== dir.charCodeAt( i ) ) {
            matches = false;
            break;
          }
        }
        if ( matches ) {
          return true;
        }
      }
    }
    return false;
  }

  check( file ) {
    this.pending++;

    this.fsStat( file ).then( stats => {

      if ( this.cancelled ) {
        return;
      }

      this.pending--;

      const relativePath = slash( path.relative( this.cwd, file ) );

      if ( stats.isDirectory() ) {
        if ( this.includeDirs && this.matches( relativePath ) ) {
          this.next( this.relative ? relativePath : file );
          this.readDir( file );
        } else if ( this.matchesDir( relativePath ) ) {
          this.readDir( file );
        }
      } else if ( stats.isFile() ) {
        if ( this.matches( relativePath ) ) {
          this.next( this.relative ? relativePath : file );
        }
      }

      this.checkPending();

    }, this.error );
  }

  readDir( folder ) {
    if ( this.visitedDirs ) {
      this.visitedDirs.push( slash( path.relative( this.cwd, folder ) ) );
    }

    this.pending++;

    this.fsReadDir( folder ).then( files => {

      if ( this.cancelled ) {
        return;
      }

      this.pending--;

      files.forEach( f => this.check( path.join( folder, f ) ) );

      this.checkPending();

    }, this.error );
  }

}

function makeSureArray( val ) {
  if ( val == null ) {
    return [];
  }
  return Array.isArray( val ) ? val : [ val ];
}

export function findFilesObservable( src, _opts ) {

  const options = Object.assign( {}, _opts );

  options.cwd = options.cwd ? path.resolve( options.cwd ) : process.cwd();
  options.fs = options.fs || fs;
  options.fs.stat = options.fs.stat || fs.stat;
  options.fs.readdir = options.fs.readdir || fs.readdir;
  options.append = makeSureArray( options.append || ignoredDirectories );

  const patterns = makeSureArray( src ).concat( options.append ).map(
    pattern => new Pattern( pattern, options )
  );

  return new Observable( observer => new Reader( patterns, observer, options ).start() );
}

export default function( patterns, options ) {
  return new Promise( ( resolve, reject ) => {
    const files = [];

    findFilesObservable( patterns, options ).subscribe( {
      error: reject,
      next: x => files.push( x ),
      complete: () => resolve( files )
    } );
  } );
}
