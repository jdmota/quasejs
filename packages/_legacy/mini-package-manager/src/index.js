// @flow
import { execOnInternal, execOnFolder, execOnBoth, copyMaybe, crawl } from "./utils";
import { before as npmBefore, after as npmAfter, normalize as npmNormalize } from "./npm";

const path = require( "path" );
const fs = require( "fs-extra" );
const crypto = require( "crypto" );
const homedir = require( "os" ).homedir();
const readPkgUp = require( "read-pkg-up" );

function connectNodeModules( deps: Set<string>, fromBase: string, toBase: string ) {
  const promises = [];

  for ( const dep of deps ) {

    const _from = path.join( fromBase, "node_modules", dep );
    const to = path.join( toBase, "node_modules", dep );

    promises.push(
      fs.pathExists( _from ).then( exists => {
        if ( exists ) {
          return crawl( _from, item => {
            if ( item.stats.isFile() ) {
              const relative = path.relative( _from, item.path );
              const dest = path.resolve( to, relative );
              return (
                /\.js$/.test( dest ) || !/\..+$/.test( dest ) ?
                  fs.outputFile( dest, `module.exports=require(${JSON.stringify( item.path )});` ) :
                  fs.copy( item.path, dest )
              );
            }
          } );
        }
      } )
    );
  }

  return Promise.all( promises );
}

function exec( command: string, args: string[], folder: string, internal: string ) {

  const thing = npmNormalize( command );

  if ( Array.isArray( thing ) ) {
    return execOnBoth( thing, args, folder, internal );
  }

  if ( thing ) {
    return execOnInternal( command, args, folder, internal );
  }

  return execOnFolder( command, args, folder );
}

// TODO https://stackoverflow.com/questions/14742553/npm-local-install-package-to-custom-location
// TODO https://github.com/yarnpkg/yarn/issues/1684

// modules in ./npm_modules/node_modules
// npm install slash --prefix ./npm_modules

// modules in ./yarn_modules
// yarn add slash --modules-folder ./yarn_modules

export default async function( command: string, args: string[] ) {

  const { pkg, path: pkgPath } = readPkgUp();

  const folder = path.join( pkgPath, ".." );
  const folderName = path.basename( folder );

  const refPath = path.join( folder, ".quase-mini-package-manager" );

  let refExisted = false;
  let ref;

  try {
    ref = await fs.readFile( refPath, "utf8" );
    refExisted = true;
  } catch ( err ) {
    if ( err.code === "ENOENT" ) {
      ref = `${folderName}-${crypto.createHash( "md5" ).update( Date.now() + "", "utf8" ).digest( "hex" )}`;
    } else {
      throw err;
    }
  }

  const internal = path.join( homedir, "quase-mini-package-manager", ref );

  await fs.ensureDir( internal );

  await copyMaybe( folder, internal, npmBefore );

  if ( !refExisted ) {
    await fs.writeFile( refPath, ref );
  }

  await exec( command, args, folder, internal );

  await copyMaybe( internal, folder, npmAfter );

  const deps = new Set();

  for ( const kind of [ "dependencies", "devDependencies", "peerDependencies", "optionalDependencies" ] ) {
    for ( const name in Object( pkg[ kind ] ) ) {
      deps.add( name );
    }
  }

  await connectNodeModules( deps, internal, folder );

}
