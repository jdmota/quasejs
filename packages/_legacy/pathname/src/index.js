import basename from "./basename";
import changeBase from "./change-base";
import dirname from "./dirname";
import extname from "./extname";
import isAbsolute from "./is-absolute";
import join from "./join";
import match from "./match";
import normalize from "./normalize";
import pathToRegExp from "./path-to-regexp";
import relative from "./relative";
import resolve from "./resolve";

export default {
  basename,
  changeBase,
  dirname,
  extname,
  isAbsolute,
  join,
  match,
  normalize,
  pathToRegExp,
  relative,
  resolve
};
