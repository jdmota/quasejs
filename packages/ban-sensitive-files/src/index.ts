import { Options } from "./types";
import isBanned from "./is-banned";
import collectFiles from "./collect-files";
import checkSensitiveFiles from "./check-sensitive-files";

function msg( number: number, string: string ): string {
  return string.replace( /%d/g, number + "" ).replace( /%s/g, number === 1 ? "" : "" );
}

export async function bin( options: Options ) {
  /* eslint-disable no-console */
  console.log( "Searching for sensitive data..." );

  const { banned, sensitive } = await run( options );

  if ( banned.length > 0 ) {
    console.error( msg( banned.length, `\n%d banned file%s found:` ) );
    for ( const b of banned ) {
      console.error( `  ${b.filename}` );
      if ( options.verbose ) {
        for ( const rule of b.rules ) {
          if ( rule.caption ) {
            console.error( `    ${rule.caption}` );
          }
          if ( rule.description ) {
            console.error( `      ${rule.description}` );
          }
        }
      }
    }
  }

  if ( sensitive.length > 0 ) {
    console.error( msg( sensitive.length, `\n%d file%s with sensitive data found:` ) );
    for ( const s of sensitive ) {
      console.error( `  ${s.filename}` );
      if ( options.verbose ) {
        for ( const error of s.errors ) {
          console.error( `    ${error}` );
        }
      }
    }
  }

  const error = banned.length > 0 || sensitive.length > 0;
  if ( error ) {
    console.error( msg( banned.length + sensitive.length, `\n%d sensitive file%s found.` ) );
    process.exitCode = 1;
  } else {
    console.log( "All ok!" );
  }
  return error;
}

async function run( options: Options, providedFilenames?: string[] ) {
  const filenames = providedFilenames || await collectFiles( options );
  const banned = isBanned( filenames );
  const sensitive = await checkSensitiveFiles( filenames );
  return {
    banned,
    sensitive
  };
}

export default run;
