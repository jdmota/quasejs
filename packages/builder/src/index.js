import Builder from "./builder";
import Watcher from "./watcher";

const { printError } = require( "@quase/config" );
const EventEmitter = require( "events" );

function run( options ) {
  let reporter, emitter;
  options = options || {};
  options.warn = options.warn || ( w => emitter.emit( "warning", w ) );

  const builder = new Builder( options );
  const { plugin: Reporter, options: reporterOpts } = builder.reporter;

  if ( builder.watch ) {
    emitter = new Watcher( builder );
    reporter = new Reporter( reporterOpts, builder, emitter );
    emitter.start();
  } else {
    emitter = new EventEmitter();
    reporter = new Reporter( reporterOpts, builder, emitter );
    emitter.emit( "build-start" );
    builder.build().then(
      o => emitter.emit( "build", o ),
      e => emitter.emit( "build-error", e )
    );
  }
  return reporter;
}

export default function( options ) {
  try {
    return run( options );
  } catch ( e ) {
    printError( e );
  }
}

export { Builder, Watcher };
