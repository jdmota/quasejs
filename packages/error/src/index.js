const stackParser = require( "error-stack-parser" );
const { prettify } = require( "@quase/path-url" );

const ignoreStackTraceRe = /StackTrace\$\$|ErrorStackParser\$\$|StackTraceGPS\$\$|StackGenerator\$\$/;
const ignoreFileRe = /^([^()\s]*\/node_modules\/quase[^()\s]+|[^()\s\\/]+\.js|internal\/[^()\s\\/]+\/[^()\s\\/]+\.js|native)$/;

export async function beautify( originalStack, extractor ) {

  const frames = stackParser.parse( { stack: originalStack } ).filter( ( { fileName, functionName } ) => {
    return !ignoreFileRe.test( fileName ) && !ignoreStackTraceRe.test( functionName || "" );
  } );

  const promises = frames.map( async( { fileName, functionName, args, lineNumber, columnNumber } ) => {

    const stackLine = {
      textLine: `${functionName}${args ? `(${args.join( ", " )})` : ""}`,
      file: fileName,
      code: null,
      line: lineNumber,
      column: columnNumber
    };

    try {
      const pos = await extractor.getOriginalLocation( stackLine.file, { line: stackLine.line, column: stackLine.column } );
      if ( pos.line == null ) {
        stackLine.code = pos.code;
      } else {
        stackLine.file = pos.originalFile;
        stackLine.code = pos.originalCode;
        stackLine.line = pos.line;
        stackLine.column = pos.column;
      }
    } catch ( e ) {
      // Ignore
    }

    return stackLine;

  } );

  const cleaned = await Promise.all( promises );
  const cleanedText = cleaned.map( ( { textLine, file, line, column } ) => ( `${textLine} (${prettify( file )}:${line}:${column})` ) );

  const title = originalStack.split( "\n" ).shift();
  const lines = cleanedText.map( x => `    ${x}` ).join( "\n" );
  const stack = `${title}\n${lines}`;

  const first = cleaned[ 0 ];

  return {
    stack,
    source: first && first.file ? first : null
  };
}

export function getStack( offset ) {
  let error = new Error();

  // Not all browsers generate the `stack` property
  // Safari <=7 only, IE <=10 - 11 only
  /* istanbul ignore if */
  if ( !error.stack ) {
    try {
      throw error;
    } catch ( err ) {
      error = err;
    }
  }

  const arr = error.stack.split( "\n" );
  arr.splice( 1, offset || 1 );
  return arr.join( "\n" );
}

let installed = false;
let prevPrepareStack;
let prevExtension;

function stripBOM( content ) {
  if ( content.charCodeAt( 0 ) === 0xFEFF ) {
    content = content.slice( 1 );
  }
  return content;
}

export function install( extractor ) {
  if ( !installed ) {
    installed = true;

    prevPrepareStack = Error.prepareStackTrace;
    Error.prepareStackTrace = function( error ) {
      return beautify( error.stack, extractor );
    };

    if ( typeof require === "function" ) {
      const Module = require( "module" );
      prevExtension = Module._extensions[ ".js" ];
      Module._extensions[ ".js" ] = function( module, filename ) {
        const content = extractor.fs.getFileSync( filename );
        module._compile( stripBOM( content ), filename );
      };
    }
  }
}

export function uninstall() {
  if ( installed ) {
    Error.prepareStackTrace = prevPrepareStack;
    if ( prevExtension ) {
      const Module = require( "module" );
      Module._extensions[ ".js" ] = prevExtension;
    }
    prevPrepareStack = undefined;
    prevExtension = undefined;
    installed = false;
  }
}
