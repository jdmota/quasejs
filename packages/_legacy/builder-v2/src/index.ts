import { Options } from "./types";
import { Builder } from "./builder/builder";
import Reporter from "./reporter";

export async function cli(options: Options) {
  const builder = new Builder(options);
  const reporter = new Reporter({}, builder); // TODO custom reporter and options

  process.once("SIGINT", () => {
    builder.emit("sigint");
    builder.stop();
  });

  const { result, firstJob } = await builder.start();

  return {
    reporter,
    builder,
    setupResult: result,
    firstJob,
  };
}

export { Builder };

export { schema, normalizeOptions } from "./options";
