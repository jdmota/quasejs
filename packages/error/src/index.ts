import stackParser from "error-stack-parser";
import { slash, prettify } from "../../util/path-url";
import { type SourceMapExtractor } from "../../source-map/src/extractor";

export const ignoreStackTraceRe =
  /StackTrace\$\$|ErrorStackParser\$\$|StackTraceGPS\$\$|StackGenerator\$\$/;
export const ignoreFileRe =
  /^([^()\s]*\/quasejs\/packages\/[^()\s/]+\/dist\/[^()\s]*|[^()\s]*\/node_modules\/@quase\/[^()\s]+|[^()\s/]+\.js|node:[^()\s]+|internal(\/[^()\s/]+)?\/[^()\s]+\.js|native)$/;

type Opts = Readonly<{
  extractor?: SourceMapExtractor;
  ignore?:
    | {
        test(text: string): boolean;
      }
    | null
    | undefined;
}>;

type StackFrame = stackParser.StackFrame & { fileName: string };

function excludeFramesWithoutFilename(
  v: stackParser.StackFrame
): v is StackFrame {
  return !!v.fileName;
}

export type BeautifiedStackLine = {
  textLine: string;
  file: string | null;
  code: string | null;
  name: string | null;
  line: number | null;
  column: number | null;
};

export async function beautify(originalStack: string = "", options: Opts = {}) {
  const extractor = options && options.extractor;
  const ignore = options && options.ignore;
  const error = { name: "", message: "", stack: originalStack };

  const originalFrames = stackParser
    .parse(error)
    .filter(excludeFramesWithoutFilename);

  const frames = originalFrames.filter(({ fileName, functionName }) => {
    const file = slash(fileName);
    if (ignore && (ignore.test(fileName) || ignore.test(file))) {
      return false;
    }
    // Electron
    if (
      file.includes(".app/Contents/Resources/electron.asar") ||
      file.includes(".app/Contents/Resources/default_app.asar")
    ) {
      return false;
    }
    return (
      !ignoreFileRe.test(file) && !ignoreStackTraceRe.test(functionName || "")
    );
  });

  const originalFirst: StackFrame | undefined = originalFrames[0];

  if (frames.length === 0 && originalFirst) {
    frames.push(originalFirst);
  }

  const promises = frames.map(
    async ({ fileName, functionName, args, lineNumber, columnNumber }) => {
      const stackLine: BeautifiedStackLine = {
        textLine: "",
        file: fileName,
        code: null,
        name: functionName ?? null,
        line: lineNumber ?? null,
        column: columnNumber ?? null,
      };

      if (extractor && stackLine.line && stackLine.column) {
        try {
          const pos = await extractor.getOriginalLocation(fileName, {
            line: stackLine.line,
            column: stackLine.column,
          });
          if ("line" in pos) {
            stackLine.file = pos.originalFile;
            stackLine.code = pos.originalCode;
            stackLine.name = pos.name || stackLine.name;
            stackLine.line = pos.line;
            stackLine.column = pos.column;
          } else {
            stackLine.code = pos.code;
          }
        } catch (e) {
          // Ignore
        }
      }

      stackLine.textLine = `${stackLine.name}${
        args ? `(${args.join(", ")})` : ""
      }`;
      return stackLine;
    }
  );

  const cleaned = await Promise.all(promises);
  const cleanedText = cleaned.map(
    ({ textLine, file, line, column }) =>
      `${textLine} (${file ? prettify(file) : "<unknown file>"}:${line}:${column})`
  );

  const title =
    originalFirst && originalFirst.source
      ? originalStack.split(originalFirst.source).shift()
      : originalStack;

  const lines = cleanedText.map(x => `    ${x}`).join("\n");
  const stack = `${title}${lines}`;

  const first = cleaned[0];

  return {
    stack,
    source: first && first.file ? first : null,
  };
}

export function getStack(offset?: number) {
  let error = new Error();

  // Not all browsers generate the `stack` property
  // Safari <=7 only, IE <=10 - 11 only
  if (!error.stack) {
    try {
      throw error;
    } catch (err: unknown) {
      error = err as Error;
    }
  }

  const stack = error.stack as string;
  const arr = stack.split("\n");
  arr.splice(1, offset != null && offset > 1 ? offset : 1);
  return arr.join("\n");
}

type Loc = Readonly<{
  line?: number | null | undefined;
  column?: number | null | undefined;
}>;

export function locToString(loc: Loc) {
  if (loc.line != null) {
    if (loc.column != null) {
      return `${loc.line}:${loc.column}`;
    }
    return `${loc.line}`;
  }
  return "";
}

// TODO use https://nodejs.org/en/blog/release/v22.9.0#new-api-to-retrieve-execution-stack-trace

export function getCallSites(traceLimit: number = 20) {
  const target: { stack: readonly NodeJS.CallSite[] } = {} as any;

  const originalPrepare = Error.prepareStackTrace;
  const originalStackLimit = Error.stackTraceLimit;

  Error.stackTraceLimit = traceLimit;
  Error.prepareStackTrace = (error, stackTraces) => stackTraces;
  Error.captureStackTrace(target, getCallSites);

  const capturedTraces = target.stack;

  Error.prepareStackTrace = originalPrepare;
  Error.stackTraceLimit = originalStackLimit;

  return capturedTraces;
}

export function getCallSite(offset = 0) {
  return getCallSites(2 + offset)[1 + offset];
}
