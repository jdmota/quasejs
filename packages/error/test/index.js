import { beautify, getStack } from "../src";

const { SourceMapExtractor } = require( "@quase/source-map" );
const fs = require( "fs-extra" );

it( "getStack", () => {

  const cwd = process.cwd();

  function clean( string ) {
    return string.split( cwd ).join( "" ).replace( /\\/g, "/" );
  }

  expect( clean( getStack() ) ).toMatchSnapshot( "undefined" );
  expect( clean( getStack( -1 ) ) ).toMatchSnapshot( "-1" );
  expect( clean( getStack( 0 ) ) ).toMatchSnapshot( "0" );
  expect( clean( getStack( 1 ) ) ).toMatchSnapshot( "1" );
  expect( clean( getStack( 2 ) ) ).toMatchSnapshot( "2" );
  expect( clean( getStack( 3 ) ) ).toMatchSnapshot( "3" );

} );

it( "beautify", async() => {

  const stack = getStack();
  const extractor = new SourceMapExtractor( fs );

  expect( ( await beautify( stack, extractor ) ).stack ).toMatchSnapshot( "dont ignore" );
  expect( ( await beautify( stack, extractor, { ignore: /node_modules/ } ) ).stack ).toMatchSnapshot( "ignore node_modules" );

} );

it( "beautify with title", async() => {

  const stack = new Error( "title" ).stack;
  const extractor = new SourceMapExtractor( fs );

  expect( ( await beautify( stack, extractor ) ).stack ).toMatchSnapshot( "dont ignore" );
  expect( ( await beautify( stack, extractor, { ignore: /node_modules/ } ) ).stack ).toMatchSnapshot( "ignore node_modules" );

} );

it( "keep at least one stack line", async() => {

  const stack = getStack( 2 );
  const extractor = new SourceMapExtractor( fs );

  expect( ( await beautify( stack, extractor, { ignore: /node_modules/ } ) ).stack ).toMatchSnapshot();

} );
