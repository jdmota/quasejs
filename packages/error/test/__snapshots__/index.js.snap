// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`beautify with title: dont ignore 1`] = `
"Error: title
    Object.<anonymous>.it (packages/error/test/index.js:35:17)
    Object.asyncFn (node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    resolve (node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    mapper (node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    promise.then (node_modules/jest-jasmine2/build/queue_runner.js:87:41)"
`;

exports[`beautify with title: ignore node_modules 1`] = `
"Error: title
    Object.<anonymous>.it (packages/error/test/index.js:35:17)"
`;

exports[`beautify: dont ignore 1`] = `
"Error
    Object.<anonymous>.it (packages/error/test/index.js:25:17)
    Object.asyncFn (node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    resolve (node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    mapper (node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    promise.then (node_modules/jest-jasmine2/build/queue_runner.js:87:41)"
`;

exports[`beautify: ignore node_modules 1`] = `
"Error
    Object.<anonymous>.it (packages/error/test/index.js:25:17)"
`;

exports[`getStack: -1 1`] = `
"Error
    at Object.<anonymous>.it (/packages/error/test/index.js:15:18)
    at Object.asyncFn (/node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    at resolve (/node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    at new Promise (<anonymous>)
    at mapper (/node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    at promise.then (/node_modules/jest-jasmine2/build/queue_runner.js:87:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`;

exports[`getStack: 0 1`] = `
"Error
    at Object.<anonymous>.it (/packages/error/test/index.js:16:18)
    at Object.asyncFn (/node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    at resolve (/node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    at new Promise (<anonymous>)
    at mapper (/node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    at promise.then (/node_modules/jest-jasmine2/build/queue_runner.js:87:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`;

exports[`getStack: 1 1`] = `
"Error
    at Object.<anonymous>.it (/packages/error/test/index.js:17:18)
    at Object.asyncFn (/node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    at resolve (/node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    at new Promise (<anonymous>)
    at mapper (/node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    at promise.then (/node_modules/jest-jasmine2/build/queue_runner.js:87:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`;

exports[`getStack: 2 1`] = `
"Error
    at Object.asyncFn (/node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    at resolve (/node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    at new Promise (<anonymous>)
    at mapper (/node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    at promise.then (/node_modules/jest-jasmine2/build/queue_runner.js:87:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`;

exports[`getStack: 3 1`] = `
"Error
    at resolve (/node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    at new Promise (<anonymous>)
    at mapper (/node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    at promise.then (/node_modules/jest-jasmine2/build/queue_runner.js:87:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`;

exports[`getStack: undefined 1`] = `
"Error
    at Object.<anonymous>.it (/packages/error/test/index.js:14:18)
    at Object.asyncFn (/node_modules/jest-jasmine2/build/jasmine_async.js:108:37)
    at resolve (/node_modules/jest-jasmine2/build/queue_runner.js:56:12)
    at new Promise (<anonymous>)
    at mapper (/node_modules/jest-jasmine2/build/queue_runner.js:43:19)
    at promise.then (/node_modules/jest-jasmine2/build/queue_runner.js:87:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)"
`;

exports[`keep at least one stack line 1`] = `
"Error
    Object.asyncFn (node_modules/jest-jasmine2/build/jasmine_async.js:108:37)"
`;
